spec/ics-002-client-semantics/README.md

```diff
@@ -378,6 +378,110 @@ type verifyNextSequenceRecv = (
   => boolean
 ```
 
+#### Query interface
+
+##### Chain queries
+
+These query endpoints are assumed to be exposed over HTTP or an equivalent RPC API by nodes of the chain associated with a particular client.
+
+`queryHeader` MUST be defined by the chain which is validated by a particular client, and should allow for retrieval of headers by height. This endpoint is assumed to be untrusted.
+
+```typescript
+type queryHeader = (
+  height: uint64
+) => Header
+```
+
+`queryChainConsensusState` MAY be defined by the chain which is validated by a particular client, to allow for the retrieval of the current consensus state which can be used to construct a new client.
+When used in this fashion, the returned `ConsensusState` MUST be manually confirmed by the querying entity, since it is subjective. This endpoint is assumed to be untrusted.
+
+```typescript
+type queryChainConsensusState = (height: uint64) => ConsensusState
```

Comment:
Might be helpful to define what is in the `ConsensusState` struct

Reply:
That's defined per-client-type. I'll add a note.